vcr/oidc4vci/service/identifiers_test.go

```diff
@@ -0,0 +1,185 @@
+
+		identifier, err := resolver.Resolve(issuerDID)
+
+		require.NoError(t, err)
+		require.Equal(t, issuerIdentifier, identifier)
+	})
+	t.Run("DID not found", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		serviceResolver := didservice.NewMockServiceResolver(ctrl)
+		serviceResolver.EXPECT().Resolve(issuerQuery, didservice.DefaultMaxServiceReferenceDepth).Return(did.Service{}, types.ErrNotFound)
+		resolver := DIDIdentifierResolver{ServiceResolver: serviceResolver}
+
+		identifier, err := resolver.Resolve(issuerDID)
+
+		require.NoError(t, err)
+		require.Empty(t, identifier)
+	})
+	t.Run("service not found", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		serviceResolver := didservice.NewMockServiceResolver(ctrl)
+		serviceResolver.EXPECT().Resolve(issuerQuery, didservice.DefaultMaxServiceReferenceDepth).Return(did.Service{}, types.ErrServiceNotFound)
+		resolver := DIDIdentifierResolver{ServiceResolver: serviceResolver}
+
+		identifier, err := resolver.Resolve(issuerDID)
+
+		require.NoError(t, err)
+		require.Empty(t, identifier)
+	})
+	t.Run("invalid service", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		serviceResolver := didservice.NewMockServiceResolver(ctrl)
+		serviceResolver.EXPECT().Resolve(issuerQuery, didservice.DefaultMaxServiceReferenceDepth).
+			Return(did.Service{ServiceEndpoint: map[string]string{"foo": "bar"}}, nil)
+		resolver := DIDIdentifierResolver{ServiceResolver: serviceResolver}
+
+		identifier, err := resolver.Resolve(issuerDID)
+
+		require.NoError(t, err)
+		require.Empty(t, identifier)
+	})
+	t.Run("other error", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		serviceResolver := didservice.NewMockServiceResolver(ctrl)
+		serviceResolver.EXPECT().Resolve(issuerQuery, didservice.DefaultMaxServiceReferenceDepth).Return(did.Service{}, errors.New("b00m!"))
+		resolver := DIDIdentifierResolver{ServiceResolver: serviceResolver}
+
+		identifier, err := resolver.Resolve(issuerDID)
+
+		require.EqualError(t, err, "unable to resolve node-http-services-baseurl service: b00m!")
+		require.Empty(t, identifier)
+	})
+}
+
+func TestTLSIdentifierResolver(t *testing.T) {
+	logrus.SetLevel(logrus.TraceLevel)
+
+	tlsConfig := &tls.Config{
+		Certificates:       []tls.Certificate{pki.Certificate()},
+		InsecureSkipVerify: true,
+	}
+	id := did.MustParseDID("did:nuts:GvkzxsezHvEc8nGhgz6Xo3jbqkHwswLmWw3CYtCm7hAW")
+
+	t.Run("ok - resolved from certificate SAN", func(t *testing.T) {
+		httpServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.Method != http.MethodHead {
+				w.WriteHeader(http.StatusMethodNotAllowed)
+				return
+			}
+			w.Header().Set("Content-Type", "application/json")
+			if strings.Contains(r.URL.Path, "/.well-known/openid-credential-issuer") {
+				w.WriteHeader(http.StatusOK)
+			} else {
+				w.WriteHeader(http.StatusNotFound)
+			}
+		}))
+		httpServer.TLS = tlsConfig.Clone()
+		httpServer.StartTLS()
```

Comment:
Does the server need TLS?

Reply:
We decided to only check port 443, so I assumed we only resolve identifier on HTTPS endpoints.